Remove outdated submodules

cbouss opened the pull request:
Remove outdated submodules - submodules pinned to specific versions, or submodules of feedstock that are inactive.

Also remove old root aggregate files (old cbcs and such).

Related script: https://github.com/anaconda/distro-incubator/pull/113

markan commented:
Thank you; appreciate you going through and removing so much debris.

ELundby45 commented:
We probably want to update our [docs](https://github.com/anaconda/perseverance-skills/blob/main/sections/02_Package_building/01_How_tos/Creating_versioned_feedstocks_from_existing_feedstocks.md?plain=1#L33-L45) if we no longer want to add submodules for versioned feedstocks.

cbouss commented:
> We probably want to update our [docs](https://github.com/anaconda/perseverance-skills/blob/main/sections/02_Package_building/01_How_tos/Creating_versioned_feedstocks_from_existing_feedstocks.md?plain=1&rgh-link-date=2025-04-29T21%3A45%3A52Z#L33-L45) if we no longer want to add submodules for versioned feedstocks.
> 
> This is removing a submodule that I've added in the past week following those docs.

It's good to add submodules for versioned feedstocks for versions we plan to maintain in parallel (e.g. multiple python versions). For others, aggregate_check is enough. I took a sledgehammer at removing these pins though. If I removed something I was not supposed to I can add it back.

cbouss commented:
https://github.com/anaconda/perseverance-skills/pull/295
